crates/sui-json-rpc/src/read_api.rs

```diff
@@ -143,6 +148,25 @@ impl ReadApiServer for ReadApi {
             .cloned()
             .map_or(cursor, |o_info| Some(o_info.object_id));
 
+        // filter here by filter options until we index tables by those options
+        match filter {
+            Some(SuiObjectDataFilter::StructType(struct_tag)) => {
+                objects.retain(|obj| obj.type_.to_string() == struct_tag.to_string())
```

Comment:
actually, for this one we should do something like
```suggestion
                objects.retain(|obj| {
                     if let ObjectType::Struct(MoveObjectType) = obj.type_ {
                        let obj_tag: StructTag = obj.type_.into();
                        // If people does not provide type_params, we will match all type_params
                        // e.g. `0x2::coin::Coin` can match `0x2::coin::Coin<0x2::sui::SUI>`
                        if !struct_tag.type_params.is_empty() &&  struct_tag.type_params != obj_tag.type_params {
                            return false;
                        }
                        return obj_tag.address == struct_tag.address && obj_tag.module == struct_tag.module && obj_tag.name == struct_tag.name;
                     }
                     return false;
                })
```

 obj.type_.address == struct_tag.address && obj.type_.module == struct_tag.module && obj.type_.name == struct_tag.name